ruby/test/Issue_2560_Test.rb

```diff
@@ -0,0 +1,168 @@
+
+class Issue_2560_Test < Minitest::Test
+    def test_Issue_2560
+        model = OpenStudio::Model::Model.new
+        story2 = OpenStudio::Model::BuildingStory.new(model)
+        
+        perimeter = OpenStudio::Point3dVector.new 
+        perimeter.push(OpenStudio::Point3d.new(4, 2, 3))
+        perimeter.push(OpenStudio::Point3d.new(4, 6, 3))
+        perimeter.push(OpenStudio::Point3d.new(18, 6, 3))
+        perimeter.push(OpenStudio::Point3d.new(18, 2, 3))
+        #OptionalSpace
+        sp1 = OpenStudio::Model::Space::fromFloorPrint(perimeter,3,model)
+        assert(sp1)
+        sp1 = sp1.get
+        sp1.setBuildingStory(story2)
+        tz1 = OpenStudio::Model::ThermalZone.new(model)
+        sp1.setThermalZone(tz1)
+        
+        story1 = OpenStudio::Model::BuildingStory.new(model)
+
+        perimeter.clear()
+        perimeter.push(OpenStudio::Point3d.new(0, 0, 0))
+        perimeter.push(OpenStudio::Point3d.new(0, 3, 0))
+        perimeter.push(OpenStudio::Point3d.new(22, 3, 0))
+        perimeter.push(OpenStudio::Point3d.new(22, 0, 0))
+        #OptionalSpace
+        sp2 = OpenStudio::Model::Space::fromFloorPrint(perimeter,3,model)
+        assert(sp2)
+        sp2 = sp2.get
+        sp2.setBuildingStory(story1)
+        tz2 = OpenStudio::Model::ThermalZone.new(model)
+        sp2.setThermalZone(tz2)
+
+        perimeter.clear()
+        perimeter.push(OpenStudio::Point3d.new(0, 3, 0))
+        perimeter.push(OpenStudio::Point3d.new(0, 5, 0))
+        perimeter.push(OpenStudio::Point3d.new(22, 5, 0))
+        perimeter.push(OpenStudio::Point3d.new(22, 3, 0))
+        #OptionalSpace
+        sp3 = OpenStudio::Model::Space::fromFloorPrint(perimeter,3,model)
+        assert(sp3)
+        sp3 = sp3.get
+        sp3.setBuildingStory(story1)
+        tz3 = OpenStudio::Model::ThermalZone.new(model)
+        sp3.setThermalZone(tz3)
+
+        perimeter.clear()
+        perimeter.push(OpenStudio::Point3d.new(0, 5, 0))
+        perimeter.push(OpenStudio::Point3d.new(0, 8, 0))
+        perimeter.push(OpenStudio::Point3d.new(22, 8, 0))
+        perimeter.push(OpenStudio::Point3d.new(22, 5, 0))
+        #OptionalSpace
+        sp4 = OpenStudio::Model::Space::fromFloorPrint(perimeter,3,model)
+        assert(sp4)
+        sp4 = sp4.get
+        sp4.setBuildingStory(story1)
+        tz4 = OpenStudio::Model::ThermalZone.new(model)
+        sp4.setThermalZone(tz4)
+
+        # Model before intersection
+        outpath = File.join(File.expand_path('../../.') + "/build/resources/model/","2560_before.osm") 
+        model.save(outpath, true)
+
+        spaces = model.getSpaces
+        #spaces = spaces.get
+        spacesX = OpenStudio::Model::SpaceVector.new
+        spaces.each do |s|
+            spacesX << s
+        end
+
+        OpenStudio::Model::intersectSurfaces(spacesX)
+        OpenStudio::Model::matchSurfaces(spacesX)
+
+        outpath = File.join(File.expand_path('../../.') + "/build/resources/model/","2560_after.osm") 
```

Comment:
I suspect this assumption will repeat. I won't comment on any remaining cases.

Reply:
https://github.com/NREL/OpenStudio/commit/fb25ab9186fbd6fb71ed041858e18380bd436ca2